tool/site/client-react/src/components/longRunningTask.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from "react"
+import {JSXElementConstructor, useContext, useEffect, useState} from "react"
+import {Button, ErrorMessage, Label} from "nhsuk-react-components"
+import ButtonList from "./buttonList"
+import {AppContext} from "../index"
+
+interface LongRunningTaskProps<T> {
+  task: () => Promise<T>
+  loadingMessage: string
+  children: JSXElementConstructor<T>
+  back?: string | (() => void)
+}
+
+const LongRunningTask = <T extends unknown>({
```

Comment:
Is it worth testing this explicitly or is it all covered in the implementations of a LongRunningTask? (i.e. covering the permutations on errors and loading messages?)

Reply:
We test it by testing the pages which use it, but yeah I'll add a more general test too